### ISSUE TITLE
Possible CVEs in Element Android Dependencies

### ISSUE THREAD

**Issue by JonasKress** [vector-im/element-android#5358]
### Steps to reproduce

Check with https://github.com/jeremylong/DependencyCheck for the Android code base.

Please see the attached reports with details.
Some of the results looked liked false positives but some of them should be fixed.

Files are attached to this ticket:
https://element-io.atlassian.net/browse/PSB-55

### Will you send logs?

No

**Comment by HarHarLinks** [comment]
dupe of #5344?

**Comment by JonasKress** [comment]
This ticket is for fixing the issue coming out of running the tool manually.
The other ticket is for setup up the check directly in the CI.

**Comment by bmarty** [comment]
Running ` ./gradlew dependencyCheckAnalyze` on current `develop` branch gives this:

```
kotlin-compiler-embeddable-1.5.20.jar (pkg:maven/org.jetbrains.kotlin/kotlin-compiler-embeddable@1.5.20, cpe:2.3:a:jetbrains:kotlin:1.5.20:*:*:*:*:*:*:*) : CVE-2022-24329
kotlin-daemon-embeddable-1.5.20.jar (pkg:maven/org.jetbrains.kotlin/kotlin-daemon-embeddable@1.5.20, cpe:2.3:a:jetbrains:kotlin:1.5.20:*:*:*:*:*:*:*) : CVE-2022-24329
kotlin-reflect-1.5.20.jar (pkg:maven/org.jetbrains.kotlin/kotlin-reflect@1.5.20, cpe:2.3:a:jetbrains:kotlin:1.5.20:*:*:*:*:*:*:*) : CVE-2022-24329
kotlin-script-runtime-1.5.20.jar (pkg:maven/org.jetbrains.kotlin/kotlin-script-runtime@1.5.20, cpe:2.3:a:jetbrains:kotlin:1.5.20:*:*:*:*:*:*:*) : CVE-2022-24329
kotlin-stdlib-1.5.20.jar (pkg:maven/org.jetbrains.kotlin/kotlin-stdlib@1.5.20, cpe:2.3:a:jetbrains:kotlin:1.5.20:*:*:*:*:*:*:*) : CVE-2022-24329
kotlin-stdlib-common-1.5.20.jar (pkg:maven/org.jetbrains.kotlin/kotlin-stdlib-common@1.5.20, cpe:2.3:a:jetbrains:kotlin:1.5.20:*:*:*:*:*:*:*) : CVE-2022-24329
kotlin-stdlib-jdk7-1.5.20.jar (pkg:maven/org.jetbrains.kotlin/kotlin-stdlib-jdk7@1.5.20, cpe:2.3:a:jetbrains:kotlin:1.5.20:*:*:*:*:*:*:*) : CVE-2022-24329
kotlin-stdlib-jdk8-1.5.20.jar (pkg:maven/org.jetbrains.kotlin/kotlin-stdlib-jdk8@1.5.20, cpe:2.3:a:jetbrains:kotlin:1.5.20:*:*:*:*:*:*:*) : CVE-2022-24329
ktlint-reporter-checkstyle-0.42.1.jar (pkg:maven/com.pinterest.ktlint/ktlint-reporter-checkstyle@0.42.1, cpe:2.3:a:checkstyle:checkstyle:0.42.1:*:*:*:*:*:*:*, cpe:2.3:a:ktlint_project:ktlint:0.42.1:*:*:*:*:*:*:*) : CVE-2019-10782, CVE-2019-9658
```

Checking all those dep one by one.

**Comment by bmarty** [comment]
After upgrading Jitsi from 3.10.0 to 5.0.1, and ktlint to 0.45.1 I got only 2 remaining possible CVE:

<pre>
ktlint-reporter-checkstyle-0.45.1.jar (pkg:maven/com.pinterest.ktlint/ktlint-reporter-checkstyle@0.45.1, cpe:2.3:a:checkstyle:checkstyle:0.45.1:*:*:*:*:*:*:*, cpe:2.3:a:ktlint_project:ktlint:0.45.1:*:*:*:*:*:*:*) : CVE-2019-10782, CVE-2019-9658
</pre>

which can be considered as a false positive, since ktlint is a static analysis tool, which does not end up in the generated binaries.

**Comment by bmarty** [comment]
Remaining issue are considered false positive and ignored.

Implemented in Element Android 1.4.10
